packages/template-retail-react-app/app/pages/product-detail/index.test.js

```diff
@@ -63,4 +264,68 @@ test('should render product details page', async () => {
     expect(screen.getAllByText(/14.99/).length).toEqual(2)
     expect(screen.getAllByText(/Add to Cart/).length).toEqual(2)
     expect(screen.getAllByText(/Add to Wishlist/).length).toEqual(2)
+    expect(screen.getAllByTestId('product-view').length).toEqual(1)
+})
+
+describe('product set', () => {
+    test('render multi-product layout', async () => {
+        renderWithProviders(<MockedPageWithProductSet />)
+
+        await waitFor(() => {
+            expect(screen.getAllByTestId('product-view').length).toEqual(4) // 1 parent + 3 children
+        })
+    })
+
+    test('add the set to cart successfully', async () => {
+        const urlPathAfterSelectingAllVariants =
+            '/en-GB/product/winter-lookM?25518447M=color%3DJJ5FUXX%26size%3D9MD&25518704M=color%3DJJ2XNXX%26size%3D9MD&25772717M=color%3DTAUPETX%26size%3D070%26width%3DM'
+        window.history.pushState({}, 'ProductDetail', urlPathAfterSelectingAllVariants)
+
+        global.server.use(
+            rest.post('*/baskets/:basketId/items', (req, res, ctx) => {
+                return res(ctx.json(basketWithProductSet))
+            })
+        )
```

Comment:
@yunakim714 did we ever decide whether this pattern (doing setup outside of a `beforeEach`) can cause race conditions. Seems like this should be in ` beforeEach` to be sure we avoid race conditions

Reply:
We can scope this server.use in a describe block to avoid it being used by other tests
```
describe('add to cart test', () => {
   beforeEach(() => {
    // this will apply to the tests inside this describe block
     global.server.use()

   })
  test('something', () => {})
})
```